### PR TITLE
Fix `packing` doc in `SFTConfig` and fix error when neither `dataset_text_field` nor `formatting_func` is provided.

### PR DESCRIPTION
**Pull request by qgallouedec**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/trl/tree/main/docs).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by qgallouedec** [comment]
Can you review @RylanSchaeffer?

**Review by lewtun — APPROVED** [review]
Thanks for the fix! LGTM but is this something that should be unit tested (happy to go with your judgement on this)?

File: trl/trainer/sft_trainer.py [hunk 1]
```diff
@@ -339,18 +339,18 @@ def make_inputs_require_grad(module, input, output):
...
-                and not args.dataset_kwargs.get("skip_prepare_dataset", False)
-            ):
-                raise ValueError(
-                    "You passed `packing=False` to the SFTTrainer/SFTConfig, but you didn't pass a `dataset_text_field` or `formatting_func` argument."
-                )
+        # If we aren't skipping data preparation, then a dataset_text_field or formatting_func must be provided.
+        if (
+            args.dataset_text_field is None
+            and formatting_func is None
+            and not args.dataset_kwargs.get("skip_prepare_dataset", False)
```
Comment (lewtun) [inline trl/trainer/sft_trainer.py:346]: Personally I find the double negatives hard to parse - would this be equivalent?
```suggestion
            and args.dataset_kwargs.get("skip_prepare_dataset", True)
```
